### PR TITLE
Seek rb fix for windows #2141 fix

### PR DESCRIPTION
**Pull request by anufdo**

### CODE REVIEW

File: seek/seek.rb [hunk 1]
```diff
@@ -296,9 +296,9 @@ def parse(args)
   is_windows = (RbConfig::CONFIG["host_os"] =~ /mswin|mingw|cygwin/)
   is_linux = (RbConfig::CONFIG["host_os"] =~ /linux/)
   if is_windows
-    system("start #{"jobs/#{filename}.csv"}")
+    exec("start \"\" \"jobs/#{filename}.csv\"")
   elsif is_linux
-    system("xdg-open #{"jobs/#{filename}.csv"}")
+    exec("xdg-open \"jobs/#{filename}.csv\"")
   else
     `open "jobs/#{filename}.csv"`
```
Comment (jbampton) [inline seek/seek.rb:303]: Needs to use `exec` here too

Comment (anufdo) [reply seek/seek.rb]: Ok.

**Review by wickedknock — COMMENTED** [review]
add error handling if the file is not found in parse method.

**Comment by BaseMax** [comment]
Hi Saad, Welcome. @wickedknock

**Comment by wickedknock** [comment]
> Hi Saad, Welcome. @wickedknock

A gun was put to my head sorry XD

**Comment by BaseMax** [comment]
> A gun was put to my head sorry XD

How do you mean?

File: seek/seek.rb [hunk 2]
```diff
@@ -296,10 +296,10 @@ def parse(args)
   is_windows = (RbConfig::CONFIG["host_os"] =~ /mswin|mingw|cygwin/)
   is_linux = (RbConfig::CONFIG["host_os"] =~ /linux/)
   if is_windows
-    exec("start \"\" \"jobs/#{filename}.csv\"")
+    exec(%(start "" "jobs/#{filename}.csv"))
   elsif is_linux
-    exec("xdg-open \"jobs/#{filename}.csv\"")
+    exec(%(xdg-open  "jobs/#{filename}.csv"))
```
Comment (jbampton) [inline seek/seek.rb:301]:
```suggestion
    exec(%(xdg-open "jobs/#{filename}.csv"))
```

Comment (anufdo) [reply seek/seek.rb:301]: OK. done

File: seek/seek.rb [hunk 3]
```diff
@@ -290,5 +291,15 @@ def parse(args)
     results.each { |row| csv_file << row }
   end
   puts "#{results.size - 1} jobs found"
-  `open "jobs/#{filename}.csv"`
+
+  # determine the current operating system
+  is_windows = (RbConfig::CONFIG["host_os"] =~ /mswin|mingw|cygwin/)
```
Comment (jbampton) [inline seek/seek.rb:296]:
```suggestion
  is_windows = (RbConfig::CONFIG["host_os"] =~ /cygwin|mingw|mswin/)
```

Comment (anufdo) [reply seek/seek.rb]: Done

**Review by BaseMax — COMMENTED** [review]
John is really great to review the code. You must learn as must as you can from him. It's +20 years of experience.

**Review by jbampton — CHANGES_REQUESTED** [review]
It would be better to use `===` instead of `=~` since `===` returns a boolean.

Also it seems that we don't need the extra `()` around `is_mac` for example or say `is_windows` etc.  Please test with Rubocop.

```
irb(main):011:0> RbConfig::CONFIG["host_os"] =~ /linux/
=> nil
irb(main):012:0> RbConfig::CONFIG["host_os"] =~ /darwin21/
=> 0
irb(main):013:0> /linux/ === RbConfig::CONFIG["host_os"]
=> false
irb(main):014:0> /darwin21/ === RbConfig::CONFIG["host_os"]
=> true
irb(main):015:0> is_mac = /darwin21/ === RbConfig::CONFIG["host_os"]
=> true
irb(main):016:0> is_mac
=> true
irb(main):017:0> /linux/ =~ RbConfig::CONFIG["host_os"]
=> nil
irb(main):018:0> /darwin21/ =~ RbConfig::CONFIG["host_os"]
=> 0
irb(main):019:0> 
```

`=~` returns the position of the match so `0` is a match at first position for say `d` in `darwin21`

Examples of `===` and `=~` here:

https://docs.ruby-lang.org/en/3.0/Regexp.html#method-i-3D-3D-3D

Note it also should be

`rxp === str → true or false` or

`rxp =~ str → integer or nil`

https://github.com/ruby/irb

**Comment by anufdo** [comment]
image

**Comment by anufdo** [comment]
It should be modified like this.
```
# determine the current operating system
host_os = RbConfig::CONFIG["host_os"]

case host_os
when /cygwin|mingw|mswin/
  exec(%(start "" "jobs/#{filename}.csv"))
when /linux/
  exec(%(xdg-open "jobs/#{filename}.csv"))
when /darwin/
  exec(%(open "jobs/#{filename}.csv"))
end
```

**Review by jbampton — COMMENTED** [review]
https://app.codacy.com/gh/slurpcode/slurp/pullRequest?prid=11506388

File: seek/seek.rb [hunk 4]
```diff
@@ -290,5 +291,18 @@ def parse(args)
     results.each { |row| csv_file << row }
   end
   puts "#{results.size - 1} jobs found"
-  `open "jobs/#{filename}.csv"`
+
+  # determine the current operating system
```
Comment (jbampton) [inline seek/seek.rb:295]:
```suggestion
```

**Review by jbampton — APPROVED** [review]

**Pull request merged**